Src/HALAL/Services/Flash/Flash.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Flash.cpp
+ *
+ *  Created on: Oct 19, 2022
+ *      Author: Pablo
+ */
+
+#include "../Inc/HALAL/Services/Flash/Flash.hpp"
+
+void Flash::read(uint32_t source_addr, uint32_t* result, uint32_t number_of_words){
+	if (source_addr < FLASH_START_ADDRESS || source_addr > FLASH_END_ADDRESS) {
+			//TODO: Handle exception (address out of memory limits)
+			return;
+	}
+
+	HAL_FLASH_Unlock();
+	while(number_of_words > 0){
+		*result = *(__IO uint32_t *)source_addr;
+		source_addr += 4;
+		result++;
+		number_of_words--;
+	}
+	HAL_FLASH_Lock();
+}
+
+//TODO: Estaria muy bien optimizar el uso de ram en la escritura de múltiples sectores
+bool Flash::write(uint32_t * source, uint32_t dest_addr, uint32_t number_of_words){
+	if (dest_addr < FLASH_SECTOR4_START_ADDRESS || dest_addr > FLASH_END_ADDRESS) {
+		//TODO: Handle exception (address out of memory limits)
+		return false;
+	}
+
+	uint32_t start_relative_position_in_words;
```

Comment:
```suggestion
	uint32_t start_relpos; // start relative position (in WORDS)
```
Debido a que hay muchas variables largas, veria mejor usar una nomenclatura de este estilo, dejaria las operaciones de abajo mucho mas limpias, pero entiendo que es bastante subjetivo.

Reply:
Por el hecho de que hayan tantas variables es porque las he hecho muy explicitas para no tener que estar subiendo a la declaración cada vez a mirar que eran exactamente.